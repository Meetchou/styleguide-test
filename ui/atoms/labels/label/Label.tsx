import React, { FC } from 'react';
import cls from 'classnames';

export type LabelProps = {
  className?: string;
  children: React.ReactNode;
  error?: boolean;
  htmlFor?: string;
};

const Label:FC<LabelProps> = ({
  className,
  children,
  error,
  htmlFor,
  ...rest
}) => (
  <label
    {...rest}
    htmlFor={htmlFor}
    className={cls(
      'block text-xs font-semibold',
      {
        'text-gray-900': !error,
        'text-danger-500': error
      },
      className
    )}
  >
    {children}
  </label>
);

export default Label;
