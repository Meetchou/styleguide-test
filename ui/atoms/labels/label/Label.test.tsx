import React from 'react';

import render from '@ui/testing/render';

import Label from './Label';

describe('atoms/labels/Label', () => {
  it('renders correctly', () => {
    const tree = render(<Label>my label</Label>);

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
