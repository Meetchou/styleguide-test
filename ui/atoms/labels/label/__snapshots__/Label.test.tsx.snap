// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atoms/labels/Label renders correctly 1`] = `
<label
  className="block text-xs font-semibold text-gray-900"
>
  my label
</label>
`;
